unyt/dask_array.py

```diff
@@ -0,0 +1,620 @@
+    # a new dask.array.core.Array object and then initialize our new unyt_dask
+    # array
+    (cls, args) = dask_array.__reduce__()
+
+    da = unyt_dask_array(
+        *args,
+        units=units,
+        registry=registry,
+        bypass_validation=bypass_validation,
+        unyt_name=unyt_name,
+    )
+
+    return da
+
+
+# note: the unyt_dask_array class has no way of catching daskified reductions.
+# operations like dask.array.min() get routed through dask.array.reductions.min()
+# and will return plain arrays or float/int values.
+#
+# When these operations exist as attributes, they can be called and will return
+# unyt objects. i.e., :
+#
+# import dask; import unyt
+# x_da = unyt_from_dask(dask.array.ones((10, 10), chunks=(2, 2)), unyt.m)
+# dask.array.min(x_da).compute()  #  returns a plain float
+# x_da.min().compute()  #  returns a unyt quantity
+#
+# but when the functions do not exist as attributes, like dask.array.nanmin(),
+# it is difficult to handle without manually wrapping all of those reductions
+# functions and exposing them here. The following function, reduce_with_units,
+# is a compromise: it is a simple helper function for calling
+# a general dask.array.reductions method on a unyt_dask array to correctly
+# handle units.
+
+_nan_ops = ["nansum", "nanmean", "nanmedian", "nanstd", "nanmax", "nanmin", "nancumsum"]
+_passthrough_reductions = ["diagonal", "median", "nanmedian"]
+
+
+def reduce_with_units(dask_func, unyt_dask_in, *args, **kwargs):
+    """
+    Call a dask.array.reduction function and preserve units.
+
+    Parameters
+    ----------
+    dask_func : function
+        a function handle from dask.array.reduction (e.g., dask.array.min,
+        dask.array.var, dask.array.nanstd) to call.
+    unyt_dask_in : unyt_dask_array
+        the unyt dask array, first argument to dask_func
+    args:
+        any arguments to the dask_func
+    kwargs:
+        any keyword arguments to the dask_func
+
+    Returns
+    -------
+    unyt_dask_array:
+        the result of dask_func with units preserved
+
+    Examples
+    --------
+    >>> from unyt import dask_array
+    >>> a = dask_array.dask.array.ones((10000,), chunks=(100,))
+    >>> a = dask_array.unyt_from_dask(a, 'm')
+    >>> b = dask_array.reduce_with_units(dask_array.dask.array.median, a, axis=0)
+    >>> b.compute()
+    unyt_quantity(1., 'm')
+
+    """
+
+    if dask_func.__name__ in _use_simple_decorator + _nan_ops + _passthrough_reductions:
+        # e.g., min, max, nanstd. functions that return the same units can
+        # be called directly as the dask function will treat unyt_dask_in as
+        # a standard dask array and then we copy over the initial units.
+        return _simple_dask_decorator(dask_func, unyt_dask_in)(
+            unyt_dask_in, *args, **kwargs
+        )
+    else:
+        # the operation may change the units
+        npfunc = getattr(np, dask_func.__name__, None)
+        if npfunc:
```

Comment:
just a personal style preference. I may be blinded by the fact that we recently gained access to the walrus operator in unyt by dropping support for Python 3.7, so please ignore this suggestion if you think it hurts readability
```suggestion
        if npfunc := getattr(np, dask_func.__name__, None)
```